src/main/scala/it/pagopa/interop/backendforfrontend/api/impl/EservicesApiServiceImpl.scala

```diff
@@ -0,0 +1,49 @@
+package it.pagopa.interop.backendforfrontend.api.impl
+
+import it.pagopa.interop.backendforfrontend.api.EservicesApiService
+import akka.http.scaladsl.marshalling.ToEntityMarshaller
+import akka.http.scaladsl.server.Route
+import it.pagopa.interop.backendforfrontend.model.{EService, Problem}
+import scala.concurrent.ExecutionContext
+import it.pagopa.interop.backendforfrontend.service.CatalogProcessService
+import it.pagopa.interop.backendforfrontend.service.TenantManagementService
+import scala.concurrent.Future
+
+import akka.http.scaladsl.server.Directives.onComplete
+import it.pagopa.interop.backendforfrontend.error.Handlers.handleError
+import scala.util.Success
+import com.typesafe.scalalogging.{Logger, LoggerTakingImplicit}
+import it.pagopa.interop.commons.logging.{CanLogContextFields, ContextFieldsToLog}
+import it.pagopa.interop.backendforfrontend.service.types.CatalogProcessServiceTypes._
+import it.pagopa.interop.commons.utils.TypeConversions._
+
+final class EservicesApiServiceImpl(
+  tenantManagementService: TenantManagementService,
+  catalogProcessService: CatalogProcessService
+)(implicit ec: ExecutionContext)
+    extends EservicesApiService {
+
+  private implicit val logger: LoggerTakingImplicit[ContextFieldsToLog] =
+    Logger.takingImplicit[ContextFieldsToLog](this.getClass)
+
+  override def getEservice(eserviceId: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerEService: ToEntityMarshaller[EService]
+  ): Route = {
+    val result: Future[EService] = for {
+      eServiceUUID    <- eserviceId.toFutureUUID
+      catalogEservice <- catalogProcessService.getEServiceById(eServiceUUID)
+      (origin, code) = (catalogEservice.producer.id, catalogEservice.producer.name)
```

Comment:
This extraction is wrong: `catalogEservice.producer.id` is not the **origin** and `catalogEservice.producer.name` is not the **code**.
This information comes from the tenant so, if I don't miss something, you should call tenantManagementService using the internal id that is the `catalogEservice.producer.id`.

Reply:
Thank you, I had in the plans to ask you about the correctness of this extraction